src/GovUk.Education.ExploreEducationStatistics.Publisher/Startup.cs

```diff
@@ -52,7 +56,22 @@ public override void Configure(IFunctionsHostBuilder builder)
                     options.UseSqlServer(ConnectionUtils.GetAzureSqlConnectionString("PublicStatisticsDb")))
                 .AddSingleton<IFileStorageService, FileStorageService>(provider =>
                     new FileStorageService(GetConfigurationValue(provider, "PublisherStorage")))
-                .AddScoped(provider => GetBlobCacheService(provider, "PublicStorage"))
+                .AddScoped(provider =>
+                {
+                    var publicBlobCacheService = GetBlobCacheService(provider, "PublicStorage");
+                    CacheAspect.Enabled = true;
+                    BlobCacheAttribute.AddService("default", publicBlobCacheService);
+                    return publicBlobCacheService;
+                })
```

Comment:
This is registered but only run when the `PublishReleaseContentFunction` is called for the first time because it has an `IBlobCacheService` dependency.

It's not very clear that if no other service depended on `IBlobCacheService` (excluding services where it's configured directly inline such as the way we set up `PublishingService` and `ContentService`) that caching would not be enabled and no caching services would be registered.

This is the case in EES-3648 where I discovered it broken after removing the DI of `IBlobCacheService` from `PublishReleaseContentFunction`.

Reply:
I left some TODO's related to this on my EES-3648 PR in https://github.com/dfe-analytical-services/explore-education-statistics/pull/3479/commits/070456b7f9a3784f9c18568c745dbd98f79f2da3 but I didn't find a good way of configuring this outside of the setup of the services.

It seems like the setup of logging happens after our custom Startup configuration, so if we try and do something like this...

```
   var serviceProvider = builder.Services
  .AddScoped...
  ...
  .BuildServiceProvider();

  ConfigureCachingAttribute(serviceProvider);

  private static void ConfigureCachingAttribute(IServiceProvider provider)
  {
      CacheAspect.Enabled = true;

      var publicBlobCacheService = GetBlobCacheService(provider, "PublicStorage");
      BlobCacheAttribute.AddService("default", publicBlobCacheService);
  }
```

then if  will fail on `provider.GetRequiredService<ILogger<BlobCacheService>>())` within `GetBlobStorageService` and `GetBlobCacheService`